dali/operators/math/expressions/broadcasting.cc

```diff
@@ -274,33 +274,165 @@ void ExpandToNDims(TensorShape<> &sh, int ndim) {
   sh = shape_cat(TensorShape<>(std::vector<int64_t>(ndim - sh.sample_dim(), 1)), sh);
 }
 
-void SimplifyShapesForBroadcasting(TensorShape<>& lhs, TensorShape<> &rhs) {
+
+SmallVector<std::pair<int, int>, 5> SimplifiedShapeCollapseGroups(span<TensorShape<>*> shapes) {
```

Comment:
I think that this function could be split into one that extracts the groups and one that collapses the shapes (we actually have the utilities for the latter).

Reply:
Actually this function was just meant to extract the groups (didn't modify the shapes).